web-scraper/src/crawler_setup.js

```diff
@@ -267,10 +267,13 @@ class CrawlerSetup {
 
         // Add initial cookies, if any.
         if (this.input.initialCookies && this.input.initialCookies.length) {
-            const cookiesToSet = tools.getMissingCookiesFromSession(session, this.input.initialCookies, request.url);
+            const cookiesToSet = session
+                ? tools.getMissingCookiesFromSession(session, this.input.initialCookies, request.url)
+                : this.input.initialCookies;
             if (cookiesToSet && cookiesToSet.length) {
                 // setting initial cookies that are not already in the session and page
-                session.setPuppeteerCookies(cookiesToSet, request.url);
+                if (session) session.setPuppeteerCookies(cookiesToSet, request.url);
+                // TODO: We have to change this when there is an option to define blocked status codes in sessionPool
                 await page.setCookie(...cookiesToSet);
```

Comment:
This makes it look like the TODO belongs with the `page.setCookie`, which is confusing. It's also not very clear from the comment what needs to be done.

Reply:
Changed.